File: src/Entity/Publication.php
<?php
namespace App\Entity;

use App\Repository\PublicationRepository;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\DBAL\Types\Types;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: PublicationRepository::class)]
class Publication
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[Assert\NotBlank(message: "Veuillez saisir le titre de la publication.")]
    #[ORM\Column(length: 255)]
    private ?string $Titre = null;

    #[Assert\NotBlank(message: "Veuillez saisir la description de la publication.")]
    #[ORM\Column(length: 255)]
    private ?string $Description = null;
    
    #[Assert\NotBlank(message: "Veuillez saisir l'URL de la ressource.")]
    #[ORM\Column(length: 255)]
    private ?string $UrlRessource = null;

    #[ORM\Column(type: 'integer')]
    private ?int $rating = 0;

    /**
     * @return int|null
     */
    public function getRating(): ?int
    {
        return $this->rating;
    }

    /**
     * @param int|null $rating
     */
    public function setRating(?int $rating): void
    {
        $this->rating = $rating;
    }

    /**
     * @return int|null
     */
    public function getSomme(): ?int
    {
        return $this->somme;
    }

    /**
     * @param int|null $somme
     */
    public function setSomme(?int $somme): void
    {
        $this->somme = $somme;
    }
    #[ORM\Column(type: 'integer')]
    private ?int $somme = 0;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): ?string
    {
        return $this->Titre;
    }

    public function setTitre(string $Titre): static
    {
        $this->Titre = $Titre;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): static
    {
        $this->Description = $Description;

        return $this;
    }

    public function getUrlRessource(): ?string
    {
        return $this->UrlRessource;
    }

    public function setUrlRessource(string $UrlRessource): static
    {
        $this->UrlRessource = $UrlRessource;

        return $this;
    }

   
}

File: src/main/java/entities/Publication.java
package entities;

public class Publication {
    private int id;
    private String titre;
    private String description;
    private String url_ressource;
    private float rating;
    private float somme;


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl_ressource() {
        return url_ressource;
    }

    public Publication(int id, String titre, String description, String url_ressource, float rating, float somme) {
        this.id = id;
        this.titre = titre;
        this.description = description;
        this.url_ressource = url_ressource;
        this.rating = rating;
        this.somme = somme;
    }

    public void setUrl_ressource(String url_ressource) {
        this.url_ressource = url_ressource;
    }

    public float getRating() {
        return rating;
    }

    public void setRating(float rating) {
        this.rating = rating;
    }

    public float getSomme() {
        return somme;
    }

    public void setSomme(float somme) {
        this.somme = somme;
    }
}

